tomlparams/tomlparams.py

```diff
@@ -264,6 +273,44 @@ def read_toml_file(self, report: bool = False, name: Optional[str] = None):
                 print('Using default parameters.')
         return outer_params
 
+    def read_defaults_as_directory(self, fullpath: str) -> dict:
+        """
+        Reads defaults from a directory of TOML files.
+
+        Args:
+            fullpath: path to directory containing TOML files
+
+        Returns:
+            dictionary of defaults
+        """
+        defaults = {}
+        toml_dict = {}
+        all_tomls = sorted(glob(f'{fullpath}/**/*.toml', recursive=True))
+        for i, toml in enumerate(all_tomls):
+            toml_dict = load_toml(toml)
+            if 'include' in toml_dict:
+                error(
+                    f'TOML file {toml} includes key "include",\nwhich'
+                    ' is not allow in the consolidated defaults.'
+                )
+            if defaults:
+                defaults_concatenated_keys = (
```

Comment:
I would make `defaults_concatenated_keys` and `toml_dict_concatenated_keys` here sets, and you could then do 
```
if keys_in_common := defaults_concatenated_keys.intersection(toml_dict_concatenated_keys):
    raise KeyError(
                            f"Duplicated key(s) '{keys_in_common}' in {toml}. Check"
                            f" any of the files in {all_tomls[:i]}"
                        )